Zilon.Core/Zilon.Core/Tactics/GameLoop.TaskState.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Zilon.Core.Tactics.Behaviour;
+
+namespace Zilon.Core.Tactics
+{
+    public sealed partial class GameLoop
+    {
+        private class TaskState
+        {
+            private readonly int _valueToExecute;
+
+            public TaskState(IActor actor, IActorTask task, IActorTaskSource taskSource)
+            {
+                Actor = actor ?? throw new ArgumentNullException(nameof(actor));
+                Task = task ?? throw new ArgumentNullException(nameof(task));
+                TaskSource = taskSource ?? throw new ArgumentNullException(nameof(taskSource));
+
+                Counter = Task.Cost;
+                _valueToExecute = Task.Cost / 2;
+            }
+
+            public IActorTask Task { get; }
+            public IActorTaskSource TaskSource { get; }
+            public int Counter { get; private set; }
+            public void UpdateCounter()
+            {
+                Counter--;
+            }
+
+            public bool TaskIsExecuting { get => Counter == _valueToExecute; }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private property 'TaskIsExecuting'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5544638)